nf_core/components/nfcore_component.py

```diff
@@ -170,65 +171,95 @@ def _get_included_components_in_chained_tests(self, main_nf_test: Union[Path, st
 
     def get_inputs_from_main_nf(self) -> None:
         """Collect all inputs from the main.nf file."""
-        inputs: list[list[dict[str, dict[str, str]]]] = []
+        inputs: Any = []  # Can be 'list[list[dict[str, dict[str, str]]]]' or 'list[str]'
```

Comment:
```suggestion
        inputs: Union[List[List[Dict[str, Dict[str, str]]]],List[str]] = []
```

Reply:
This wasn't working, because we append one of the types later on :( I learnt about the [overload](https://stackoverflow.com/questions/59359943/how-to-write-typing-overload-decorator-for-bool-arguments-by-value) decorator, but in our case it's not a function.